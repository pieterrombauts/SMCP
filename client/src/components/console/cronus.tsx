import React from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav'
import Tab from 'react-bootstrap/Tab'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import cronus_1 from "./../../media/Cronus_1.png"
import cronus_2 from "./../../media/Cronus_2.png"
import cronus_3 from "./../../media/Cronus_3.png"
import cronus_4 from "./../../media/Cronus_4.png"
import cronus_5 from "./../../media/Cronus_5.png"

interface AppProps {
  className?: string;
};

const cronus: React.FC<AppProps> = (props) => {
  return (
    <Tab.Container id="cronus-tabs" defaultActiveKey="comp-net">
      <Row className={props.className}>
        <Col id="cronus-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="comp-net">
              <img src={cronus_1} style={{ width: "1200px" }} />
            </Tab.Pane>
            <Tab.Pane eventKey="uhf-comms">
            <img src={cronus_2} style={{width:"1200px",position:"absolute"}}/>  
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="50" y="350" rx="5" ry="5" width="460" height="182" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="6%" y="38%" fill="white" font-size="25" font-family="Verdana">Space to Space Station</text>

                <text x="8%" y="43%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
                <text x="25%" y="43%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="8%" y="46%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
                <text x="25%" y="46%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="6%" y="51%" fill="white" font-size="25" font-family="Verdana">UHF Sync Lcok</text>
                <text x="28%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">Frame Sync</text>
                
              </svg>

              <svg width="100%" height="100%" style={{position:"absolute"}}>
                
                <rect x="740" y="370" rx="5" ry="5" width="390" height="135" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="62%" y="46%" fill="white" font-size="20" font-family="Verdana">IAC1 Power</text>
                <text x="75%" y="46%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

                <text x="62%" y="49%" fill="white" font-size="20" font-family="Verdana">IAC2 Power</text>
                <text x="75%" y="49%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

              </svg>
            </Tab.Pane>
            
            <Tab.Pane eventKey="s-band-comms">
            <img src={cronus_3} style={{width:"1200px",position:"absolute"}}/>  
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="415" y="265" rx="5" ry="5" width="415" height="385" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
                <text x="42.5%" y="31.5%" fill="white" font-size="45" font-family="Verdana">S-Band Status</text>

                <text x="42.5%" y="35%" fill="white" font-size="20" font-family="Verdana">Active S-Band</text>
                <text x="65%" y="35%" fill="#d6ab62" font-size="20" font-family="Verdana">1</text>

                <text x="42.5%" y="38%" fill="white" font-size="20" font-family="Verdana">S-Band1 Power</text>
                <text x="65%" y="38%" fill="#d6ab62" font-size="20" font-family="Verdana">On</text>

                <text x="42.5%" y="41%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="65%" y="41%" fill="#d6ab62" font-size="20" font-family="Verdana">59.14</text>
                <text x="72%" y="41%" fill="white" font-size="20" font-family="Verdana">°</text>

                <text x="42.5%" y="44%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
                <text x="65%" y="44%" fill="#d6ab62" font-size="20" font-family="Verdana">1.59</text>
                <text x="72%" y="44%" fill="white" font-size="20" font-family="Verdana">°</text>


                <text x="42.5%" y="47%" fill="white" font-size="20" font-family="Verdana">S-Band2 Power</text>
                <text x="65%" y="47%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

                <text x="42.5%" y="50%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="65%" y="50%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
                <text x="72%" y="50%" fill="white" font-size="20" font-family="Verdana">°</text>


                <text x="42.5%" y="53%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
                <text x="65%" y="53%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
                <text x="72%" y="53%" fill="white" font-size="20" font-family="Verdana">°</text>
              </svg>
              
            </Tab.Pane>

            <Tab.Pane eventKey="vid-comms-1">
              <img src={cronus_4} style={{ position:"absolute", width: "1200px" }} />
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="415" y="265" rx="5" ry="5" width="405" height="375" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
                
                <text x="42.5%" y="31.5%" fill="white" font-size="45" font-family="Verdana">Ku-Band Status</text>

                <text x="42.5%" y="35%" fill="white" font-size="20" font-family="Verdana">Downlink 1 Status</text>
                <text x="65%" y="35%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="38%" fill="white" font-size="20" font-family="Verdana">Downlink 2 Status</text>
                <text x="65%" y="38%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="41%" fill="white" font-size="20" font-family="Verdana">Downlink 3 Status</text>
                <text x="65%" y="41%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>

                <text x="42.5%" y="44%" fill="white" font-size="20" font-family="Verdana">Downlink 4 Status</text>
                <text x="65%" y="44%" fill="#d6ab62" font-size="20" font-family="Verdana">ACTIVE</text>


                <text x="42.5%" y="55%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
                <text x="67%" y="55%" fill="#d6ab62" font-size="20" font-family="Verdana">-48.85</text>
                <text x="75%" y="55%" fill="white" font-size="20" font-family="Verdana">°</text>

                <text x="42.5%" y="58%" fill="white" font-size="20" font-family="Verdana">Cross-Elevation Gimbal</text>
                <text x="67%" y="58%" fill="#d6ab62" font-size="20" font-family="Verdana">36.39</text>
                <text x="75%" y="58%" fill="white" font-size="20" font-family="Verdana">°</text>
              </svg>

            </Tab.Pane>
            
            
            <Tab.Pane eventKey="vid-comms-2">
            <img src={cronus_5} style={{ position:"absolute", width: "1200px" }} />
              <svg width="1000" height="1000" style={{position:"absolute"}}>
                
                <rect x="570" y="550" rx="5" ry="5" width="370" height="160" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
                
                <text x="58%" y="58%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="58%" fill="#d6ab62" font-size="20" font-family="Verdana">P1 Upper Outboard (P1UPOB)</text>

                <text x="58%" y="62%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="62%" fill="#d6ab62" font-size="20" font-family="Verdana">P1 Lower Outboard (P1LOOB)</text>

                <text x="58%" y="66%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="66%" fill="#d6ab62" font-size="20" font-family="Verdana">Labs Starboard (LABS)</text>

                <text x="58%" y="70%" fill="white" font-size="17" font-family="Verdana">1)</text>
                <text x="61%" y="70%" fill="#d6ab62" font-size="20" font-family="Verdana">0E</text>

                
                
              </svg>
            </Tab.Pane>
          
          </Tab.Content>
        </Col>
        <Col id="cronus-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="comp-net">COMPUTER NETWORK</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="uhf-comms">UHF COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="s-band-comms">S-BAND COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="vid-comms-1">VIDEO COMMS 1</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link eventKey="vid-comms-2">VIDEO COMMS 2</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>

    //  <Tabs  defaultActiveKey="Cronus_console" id="uncontrolled-tab-example" style={{ float: "right"}} >

    //   <Tab eventKey="CRONUS_1" title="Computer Network">
    //     <img src={cronus_1} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_2" title="UHF Communication">
    //     <img src={cronus_2} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_3" title="S-Band Communication">
    //     <img src={cronus_3} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_4" title="Video Communication">
    //     <img src={cronus_4} style={{width:"300px"}}/> 
    //   </Tab>

    //   <Tab eventKey="CRONUS_5" title="Video Communication 2">
    //     <img src={cronus_5} style={{width:"300px"}}/> 
    //   </Tab>


    // </Tabs>
  );
}

export const Cronus = styled(cronus)`
  width: 100%;
  height: 100%;
  position: absolute;

  #cronus-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #cronus-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #cronus-buttons .nav-item a {
    padding: 20px 30px;
  }

`;