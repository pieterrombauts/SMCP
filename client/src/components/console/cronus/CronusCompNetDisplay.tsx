import React from 'react';
import styled from 'styled-components';
import { cronusCompNetDisplay } from '../customTypes'

interface CronusCNDProps {
  className?: string;
  values: cronusCompNetDisplay;
};

const UCronusCompNetDisplay: React.FC<CronusCNDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_1.png'></image>
        <rect x="1055" y="180" rx="5" ry="5" width="200" height="50" style={{fill:"#000000"}} />
        <rect x="1110" y="230" rx="5" ry="5" width="100" height="50" style={{fill:"#000000"}} />
        <rect x="1110" y="250" rx="5" ry="5" width="100" height="50" style={{fill:"#000000"}} />
        <text x="1080" y="217" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.time}</text>
        <text x="1130" y="257" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.standardCmds}</text>
        <text x="1100" y="295" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.compNet.loadCmds}</text>
      </svg>  
    </div>
  )
};

const CronusCompNetDisplay = styled(UCronusCompNetDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusCompNetDisplay;