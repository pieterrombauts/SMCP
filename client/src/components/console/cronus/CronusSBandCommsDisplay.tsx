import React from 'react';
import styled from 'styled-components';
import { cronusSBandDisplay } from '../customTypes'

interface CronusSCDProps {
  className?: string;
  values: cronusSBandDisplay;
};

const UCronusSBandCommsDisplay: React.FC<CronusSCDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_3.png'></image>
        <rect x="450" y="380" rx="5" ry="5" width="450" height="385" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
        <text x="38%" y="45%" fill="white" font-size="45" font-family="Verdana">S-Band Status</text>

        <text x="38%" y="51%" fill="white" font-size="20" font-family="Verdana">Active S-Band</text>
        <text x="58%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">1</text>

        <text x="38%" y="56%" fill="white" font-size="20" font-family="Verdana">S-Band1 Power</text>
        <text x="58%" y="56%" fill="#d6ab62" font-size="20" font-family="Verdana">On</text>

        <text x="41%" y="59%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
        <text x="58%" y="59%" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.sBand.elevationGimbal}</text>
        <text x="63%" y="59%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="41%" y="62%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
        <text x="58%" y="62%" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.sBand.azimuthGimbal}</text>
        <text x="63%" y="62%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="38%" y="67%" fill="white" font-size="20" font-family="Verdana">S-Band2 Power</text>
        <text x="58%" y="67%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="41%" y="70%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
        <text x="58%" y="70%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
        <text x="63%" y="70%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="41%" y="73%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
        <text x="58%" y="73%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
        <text x="63%" y="73%" fill="white" font-size="20" font-family="Verdana">째</text>
      </svg>
    </div>
  )
};

const CronusSBandCommsDisplay = styled(UCronusSBandCommsDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusSBandCommsDisplay;