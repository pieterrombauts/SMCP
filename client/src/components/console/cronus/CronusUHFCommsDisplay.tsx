import React from 'react';
import styled from 'styled-components';

interface CronusUCDProps {
  className?: string;
};

const UCronusUHFCommsDisplay: React.FC<CronusUCDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_2.png'></image>
        
        <rect x="50" y="405" rx="5" ry="5" width="510" height="200" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
        
        <text x="6%" y="45%" fill="white" font-size="25" font-family="Verdana">Space to Space Station</text>

        <text x="8%" y="48%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
        <text x="25%" y="48%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="8%" y="51%" fill="white" font-size="20" font-family="Verdana">Radio1 Power</text>
        <text x="25%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="6%" y="57%" fill="white" font-size="25" font-family="Verdana">UHF Sync Lcok</text>
        <text x="28%" y="57%" fill="#d6ab62" font-size="20" font-family="Verdana">Frame Sync</text>
        
        <rect x="815" y="435" rx="5" ry="5" width="410" height="140" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
        
        <text x="65%" y="49%" fill="white" font-size="20" font-family="Verdana">IAC1 Power</text>
        <text x="78%" y="49%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>

        <text x="65%" y="53%" fill="white" font-size="20" font-family="Verdana">IAC2 Power</text>
        <text x="78%" y="53%" fill="#d6ab62" font-size="20" font-family="Verdana">Active</text>
      </svg>
    </div>
  )
};

const CronusUHFCommsDisplay = styled(UCronusUHFCommsDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusUHFCommsDisplay;