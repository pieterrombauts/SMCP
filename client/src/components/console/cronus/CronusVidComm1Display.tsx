import React from 'react';
import styled from 'styled-components';
import { cronusVComm1Display } from '../customTypes'

interface CronusVC1DProps {
  className?: string;
  values: cronusVComm1Display;
};

const UCronusVidComm1Display: React.FC<CronusVC1DProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_4.png'></image>
        <rect x="450" y="375" rx="5" ry="5" width="435" height="335" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>

        <text x="37%" y="45%" fill="white" fontSize="45" fontFamily="Verdana">Ku-Band Status</text>

        <text x="37%" y="50%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 1 Status</text>
        <text x="55%" y="50%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="53%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 2 Status</text>
        <text x="55%" y="53%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="56%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 3 Status</text>
        <text x="55%" y="56%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>

        <text x="37%" y="59%" fill="white" fontSize="20" fontFamily="Verdana">Downlink 4 Status</text>
        <text x="55%" y="59%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">ACTIVE</text>


        <text x="37%" y="64%" fill="white" fontSize="20" fontFamily="Verdana">Elevation Gimbal</text>
        <text x="58%" y="64%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">{props.values.vComms1.elevationGimbal}</text>
        <text x="63%" y="64%" fill="white" fontSize="20" fontFamily="Verdana">°</text>

        <text x="37%" y="67%" fill="white" fontSize="20" fontFamily="Verdana">Cross-Elevation Gimbal</text>
        <text x="58%" y="67%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">{props.values.vComms1.crossElevationGimbal}</text>
        <text x="63%" y="67%" fill="white" fontSize="20" fontFamily="Verdana">°</text>
      </svg>
    </div>
  )
};

const CronusVidComm1Display = styled(UCronusVidComm1Display)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusVidComm1Display;