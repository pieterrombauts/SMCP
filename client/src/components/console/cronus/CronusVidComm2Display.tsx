import React from 'react';
import styled from 'styled-components';

interface CronusVC2DProps {
  className?: string;
};

const UCronusVidComm2Display: React.FC<CronusVC2DProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_5.png'></image>
        <rect x="650" y="630" rx="5" ry="5" width="370" height="160" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}}/>
        
        <text x="51%" y="68%" fill="white" fontSize="17" fontFamily="Verdana">1)</text>
        <text x="54%" y="68%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">P1 Upper Outboard (P1UPOB)</text>

        <text x="51%" y="71%" fill="white" fontSize="17" fontFamily="Verdana">2)</text>
        <text x="54%" y="71%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">P1 Lower Outboard (P1LOOB)</text>

        <text x="51%" y="74%" fill="white" fontSize="17" fontFamily="Verdana">3)</text>
        <text x="54%" y="74%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">Labs Starboard (LABS)</text>

        <text x="51%" y="77%" fill="white" fontSize="17" fontFamily="Verdana">4)</text>
        <text x="54%" y="77%" fill="#d6ab62" fontSize="20" fontFamily="Verdana">0E</text>  
      </svg>
    </div>
  )
};

const CronusVidComm2Display = styled(UCronusVidComm2Display)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusVidComm2Display;