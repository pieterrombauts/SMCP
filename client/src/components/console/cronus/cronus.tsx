import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav'
import Tab from 'react-bootstrap/Tab'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import CronusCompNetDisplay from './CronusCompNetDisplay'
import CronusUHFCommsDisplay from './CronusUHFCommsDisplay'
import CronusSBandCommsDisplay from './CronusSBandCommsDisplay'
import CronusVidComm1Display from './CronusVidComm1Display'
import CronusVidComm2Display from './CronusVidComm2Display'
import { generateRandVal } from '../../../dataGenerator';
import socket from '../../Socket';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from '../../../reducers';
import { decreaseValue } from '../../../dataDecreaser';

interface AppProps {
  className?: string;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time,
})

const connector = connect(mapState)
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & AppProps

const defaultValues = {
  compNet: {
    time: '',
    standardCmds: 10466,
    loadCmds: 11248
  },
  sBand: {
    elevationGimbal: 60,
    azimuthGimbal: 2
  },
  vComms1: {
    elevationGimbal: 48,
    crossElevationGimbal: 45
  }
}

const Cronus: React.FC<Props> = ( props ) => {
  var [values, setValues] = useState(defaultValues);
  const intervalRef = useRef(0);
  useEffect(() => {
    if (intervalRef.current !== 0) {
      clearInterval(intervalRef.current);
    }
    const id = setInterval(() => {
      var newValues = {
        compNet: {
          time: "",
          standardCmds: 10466,
          loadCmds: 11248
        },
        sBand: {
          elevationGimbal:  parseFloat(decreaseValue(values.sBand.elevationGimbal).toFixed(2)),
          azimuthGimbal: parseFloat(decreaseValue(values.sBand.azimuthGimbal).toFixed(2))
        },
        vComms1: {
          elevationGimbal: parseFloat(decreaseValue(values.vComms1.elevationGimbal).toFixed(2)),
          crossElevationGimbal: parseFloat(decreaseValue(values.vComms1.crossElevationGimbal).toFixed(2))
        }
      }
      setValues(values = newValues);
    }, 5000);
    intervalRef.current = id;
  },[])
  const [ firstOpensCronus, setFirstOpensCronus ] = useState<{uhf: boolean, sband: boolean, vc1: boolean, vc2: boolean}>({uhf: true, sband: true, vc1: true, vc2: true});
  function updateFirstConsoleOpenCronus(event: React.MouseEvent<any>) {
    switch(event.currentTarget.id) {
      case 'cronus-tabs-tab-uhf-comms':
        if (firstOpensCronus.uhf) {
          socket.emit('FIRST_CONSOLE_OPEN', 'cronus-uhf')
          var newFirstOpens = firstOpensCronus;
          newFirstOpens['uhf'] = false;
          setFirstOpensCronus(newFirstOpens);
        }
        break;
      case 'cronus-tabs-tab-s-band-comms':
        if (firstOpensCronus.sband) {
          socket.emit('FIRST_CONSOLE_OPEN', 'cronus-sband')
          var newFirstOpens = firstOpensCronus;
          newFirstOpens['sband'] = false;
          setFirstOpensCronus(newFirstOpens);
        }
        break;
      case 'cronus-tabs-tab-vid-comms-1':
        if (firstOpensCronus.vc1) {
          socket.emit('FIRST_CONSOLE_OPEN', 'cronus-vc1')
          var newFirstOpens = firstOpensCronus;
          newFirstOpens['vc1'] = false;
          setFirstOpensCronus(newFirstOpens);
        }
        break;
      case 'cronus-tabs-tab-vid-comms-2':
        if (firstOpensCronus.vc2) {
          socket.emit('FIRST_CONSOLE_OPEN', 'cronus-vc2')
          var newFirstOpens = firstOpensCronus;
          newFirstOpens['vc2'] = false;
          setFirstOpensCronus(newFirstOpens);
        }
        break;
    }
  }
  return(
    <Tab.Container id="cronus-tabs" defaultActiveKey="comp-net">
      <Row className={props.className}>
        <Col id="cronus-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="comp-net">
              <CronusCompNetDisplay values={values} time={props.time} />
            </Tab.Pane>
            <Tab.Pane eventKey="uhf-comms">
              <CronusUHFCommsDisplay />
            </Tab.Pane>
            <Tab.Pane eventKey="s-band-comms">
              <CronusSBandCommsDisplay values={values} />
            </Tab.Pane>
            <Tab.Pane eventKey="vid-comms-1">
              <CronusVidComm1Display values={values} />
            </Tab.Pane>
            <Tab.Pane eventKey="vid-comms-2">
              <CronusVidComm2Display />
            </Tab.Pane>
          </Tab.Content>
        </Col>
        <Col id="cronus-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="comp-net">COMPUTER NETWORK</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleOpenCronus} eventKey="uhf-comms">UHF COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleOpenCronus} eventKey="s-band-comms">S-BAND COMMS</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleOpenCronus} eventKey="vid-comms-1">VIDEO COMMS 1</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleOpenCronus} eventKey="vid-comms-2">VIDEO COMMS 2</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>
  );
}

const ConCronus1 = styled(Cronus)`
  width: 100%;
  height: 100%;
  position: absolute;

  #cronus-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #cronus-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #cronus-buttons .nav-item a {
    padding: 20px 30px;
  }

`;

export const Cronus1 = connector(ConCronus1);
