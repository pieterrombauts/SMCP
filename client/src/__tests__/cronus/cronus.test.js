import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { Cronus } from '../../components/console/cronus/cronus'
import { shallow, mount, render} from 'enzyme';

describe("Cronus display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Cronus />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})