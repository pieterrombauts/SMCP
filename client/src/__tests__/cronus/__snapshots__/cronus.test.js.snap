// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus display renders correctly 1`] = `
<TabContainer
  defaultActiveKey="comp-net"
  id="cronus-tabs"
>
  <Row
    noGutters={false}
  >
    <Col
      id="cronus-content"
      sm={10}
    >
      <ForwardRef>
        <TabPane
          eventKey="comp-net"
        >
          <Styled(UCronusCompNetDisplay)
            values={
              Object {
                "compNet": Object {
                  "loadCmds": 11248,
                  "standardCmds": 10466,
                  "time": "",
                },
                "sBand": Object {
                  "azimuthGimbal": 2,
                  "elevationGimbal": 60,
                },
                "vComms1": Object {
                  "crossElevationGimbal": 45,
                  "elevationGimbal": 48,
                },
              }
            }
          />
        </TabPane>
        <TabPane
          eventKey="uhf-comms"
        >
          <Styled(UCronusUHFCommsDisplay) />
        </TabPane>
        <TabPane
          eventKey="s-band-comms"
        >
          <Styled(UCronusSBandCommsDisplay)
            values={
              Object {
                "compNet": Object {
                  "loadCmds": 11248,
                  "standardCmds": 10466,
                  "time": "",
                },
                "sBand": Object {
                  "azimuthGimbal": 2,
                  "elevationGimbal": 60,
                },
                "vComms1": Object {
                  "crossElevationGimbal": 45,
                  "elevationGimbal": 48,
                },
              }
            }
          />
        </TabPane>
        <TabPane
          eventKey="vid-comms-1"
        >
          <Styled(UCronusVidComm1Display)
            values={
              Object {
                "compNet": Object {
                  "loadCmds": 11248,
                  "standardCmds": 10466,
                  "time": "",
                },
                "sBand": Object {
                  "azimuthGimbal": 2,
                  "elevationGimbal": 60,
                },
                "vComms1": Object {
                  "crossElevationGimbal": 45,
                  "elevationGimbal": 48,
                },
              }
            }
          />
        </TabPane>
        <TabPane
          eventKey="vid-comms-2"
        >
          <Styled(UCronusVidComm2Display) />
        </TabPane>
      </ForwardRef>
    </Col>
    <Col
      id="cronus-buttons"
      sm={2}
    >
      <Nav
        className="flex-column"
        fill={false}
        justify={false}
        variant="pills"
      >
        <NavItem>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            eventKey="comp-net"
          >
            COMPUTER NETWORK
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            eventKey="uhf-comms"
            onClick={[Function]}
          >
            UHF COMMS
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            eventKey="s-band-comms"
            onClick={[Function]}
          >
            S-BAND COMMS
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            eventKey="vid-comms-1"
            onClick={[Function]}
          >
            VIDEO COMMS 1
          </NavLink>
        </NavItem>
        <NavItem>
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "SafeAnchor",
                "render": [Function],
              }
            }
            disabled={false}
            eventKey="vid-comms-2"
            onClick={[Function]}
          >
            VIDEO COMMS 2
          </NavLink>
        </NavItem>
      </Nav>
    </Col>
  </Row>
</TabContainer>
`;
